# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.cpp
# Commit: d07def1365a4
# Full Hash: d07def1365a49dc8ecbf5b6c10fdfbd5ec750932
# Author: pstanciu <pstanciu@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing WPT failures @lfedcm-identity.create-store-collect.tentative.sub.https.html
#   
#   This reverts commit 03d4474e360f28be0fb075379f4e4535bd34b252.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4e25ddaadc86 -r d07def1365a4 dom/ipc/WindowGlobalParent.cpp
--- a/dom/ipc/WindowGlobalParent.cpp	Wed Jun 11 13:42:13 2025 +0000
+++ b/dom/ipc/WindowGlobalParent.cpp	Wed Jun 11 18:04:27 2025 +0300
@@ -28,7 +28,6 @@
 #include "mozilla/dom/MediaController.h"
 #include "mozilla/dom/NavigatorLogin.h"
 #include "mozilla/dom/WebAuthnTransactionParent.h"
-#include "mozilla/dom/WebIdentityParent.h"
 #include "mozilla/dom/WindowGlobalChild.h"
 #include "mozilla/dom/ChromeUtils.h"
 #include "mozilla/dom/UseCounterMetrics.h"
@@ -1448,6 +1447,76 @@
   return IPC_OK();
 }
 
+IPCResult WindowGlobalParent::RecvGetIdentityCredential(
+    IdentityCredentialRequestOptions&& aOptions,
+    const CredentialMediationRequirement& aMediationRequirement,
+    bool aHasUserActivation, const GetIdentityCredentialResolver& aResolver) {
+  IdentityCredential::GetCredentialInMainProcess(
+      DocumentPrincipal(), this->BrowsingContext(), std::move(aOptions),
+      aMediationRequirement, aHasUserActivation)
+      ->Then(
+          GetCurrentSerialEventTarget(), __func__,
+          [aResolver](const IPCIdentityCredential& aResult) {
+            return aResolver({Some(aResult), NS_OK});
+          },
+          [aResolver](nsresult aErr) {
+            aResolver({Maybe<IPCIdentityCredential>(Nothing()), aErr});
+          });
+  return IPC_OK();
+}
+
+IPCResult WindowGlobalParent::RecvStoreIdentityCredential(
+    const IPCIdentityCredential& aCredential,
+    const StoreIdentityCredentialResolver& aResolver) {
+  IdentityCredential::StoreInMainProcess(DocumentPrincipal(), aCredential)
+      ->Then(
+          GetCurrentSerialEventTarget(), __func__,
+          [aResolver](const bool& aResult) { aResolver(NS_OK); },
+          [aResolver](nsresult aErr) { aResolver(aErr); });
+  return IPC_OK();
+}
+
+IPCResult WindowGlobalParent::RecvDisconnectIdentityCredential(
+    const IdentityCredentialDisconnectOptions& aOptions,
+    const DisconnectIdentityCredentialResolver& aResolver) {
+  IdentityCredential::DisconnectInMainProcess(DocumentPrincipal(), aOptions)
+      ->Then(
+          GetCurrentSerialEventTarget(), __func__,
+          [aResolver](const bool& aResult) { aResolver(NS_OK); },
+          [aResolver](nsresult aErr) { aResolver(aErr); });
+  return IPC_OK();
+}
+
+IPCResult WindowGlobalParent::RecvPreventSilentAccess(
+    const PreventSilentAccessResolver& aResolver) {
+  nsIPrincipal* principal = DocumentPrincipal();
+  if (principal) {
+    nsCOMPtr<nsIPermissionManager> permissionManager =
+        components::PermissionManager::Service();
+    if (permissionManager) {
+      permissionManager->RemoveFromPrincipal(
+          principal, "credential-allow-silent-access"_ns);
+      aResolver(NS_OK);
+      return IPC_OK();
+    }
+  }
+
+  aResolver(NS_ERROR_NOT_AVAILABLE);
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult WindowGlobalParent::RecvSetLoginStatus(
+    LoginStatus aStatus, const SetLoginStatusResolver& aResolver) {
+  nsIPrincipal* principal = DocumentPrincipal();
+  if (!principal) {
+    aResolver(NS_ERROR_DOM_NOT_ALLOWED_ERR);
+    return IPC_OK();
+  }
+  nsresult rv = NavigatorLogin::SetLoginStatus(principal, aStatus);
+  aResolver(rv);
+  return IPC_OK();
+}
+
 IPCResult WindowGlobalParent::RecvGetStorageAccessPermission(
     bool aIncludeIdentityCredential,
     GetStorageAccessPermissionResolver&& aResolve) {
@@ -1471,7 +1540,8 @@
 
   if (aIncludeIdentityCredential) {
     bool canCollect;
-    rv = identity::CanSilentlyCollect(topPrincipal, principal, &canCollect);
+    rv = IdentityCredential::CanSilentlyCollect(topPrincipal, principal,
+                                                &canCollect);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       aResolve(nsIPermissionManager::UNKNOWN_ACTION);
       return IPC_OK();
@@ -1754,11 +1824,6 @@
   return MakeAndAddRef<WebAuthnTransactionParent>();
 }
 
-already_AddRefed<PWebIdentityParent>
-WindowGlobalParent::AllocPWebIdentityParent() {
-  return MakeAndAddRef<WebIdentityParent>();
-}
-
 NS_IMPL_CYCLE_COLLECTION_CLASS(WindowGlobalParent)
 
 NS_IMPL_CYCLE_COLLECTION_UNLINK_BEGIN_INHERITED(WindowGlobalParent,